### DISCUSSION TITLE
not able to use trunc() in a timestamp column

### DISCUSSION THREAD

**Discussion by eponcede** [dbeaver/dbeaver#34903] [Q&A] [answered]
when I use oracle sql developer with the same query the funciton is working, hower in DBeaver it is not working. Any ideas, I have tried display formats and Data Formats but no luck. 
DBeaver
![image](https://github.com/user-attachments/assets/6e96c932-e7f4-4716-8a7c-d17c485dbb50)

SQL Developer
![image](https://github.com/user-attachments/assets/4bff33b6-0cc8-40cf-9b53-64e5930b4427)

**Comment by TB-HunBoka** [comment]
Good news, the trunc() function works as the time part is 00:00:00.. it's only visual. Try the same with sysdate.

Dbeaver formats values based on type, and even if you truncate datetime it is still datetime in both DB application.
On monday I can check, but I think you are looking for the to_date() function. Give it a try.
2 digit year: to_date(LAST_UPDATE_DATE, DD-MM-YY)
4 digit year: to_date(LAST_UPDATE_DATE, DD-MM-YYYY)

**Reply by TB-HunBoka** [reply, marked as answer]
I was wrong. The formatting is based on JDBC type.
The jdbc driver will map date as timestamp because the oracle date type has time values in it while JDBC date doesn't. So you will lose the time part of the date while transforming it. You can force to hide it, but be careful with it.

You can change it here:
![image](https://github.com/user-attachments/assets/cf55cbaa-8c36-4491-b474-933ff78ba14e)

Enabling this setting will result in the following:
select creation_date, trunc(creation_date) truncated, to_date(trunc(CREATION_DATE), 'DD-MM-YY') to_date_and_truncated, cast(CREATION_DATE AS DATE) casted, sysdate, cast(SYSDATE AS TIMESTAMP) from table
![image](https://github.com/user-attachments/assets/911c5242-8eb1-44ac-bc59-6ac7a19c4f7e)

You can see that the sysdate value returns an oracle date type, therefore you will not see the time value on it unless you change it's type to timestamp.

**Comment by eponcede** [comment]
One last question, how do I open the window to edit the JDBC driver?

**Reply by TB-HunBoka** [reply]
If you mean the window I print screened, then right click on the connection, edit connection.
If you mean the advanced jdbc driver settings, then Database -> Driver manager.

**Reply by eponcede** [reply]
THANK YOU...Settings are working correctly now and TRUNC()

![image](https://github.com/user-attachments/assets/136cf83c-26dd-4388-be6d-bd14d7bd81bc)
